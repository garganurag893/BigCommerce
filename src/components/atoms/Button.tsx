import { ButtonHTMLAttributes, FC } from "react";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  isLoading?: boolean;
}

const Button: FC<ButtonProps> = ({
  text,
  className,
  disabled,
  isLoading = false,
  ...props
}) => (
  <button
    disabled={disabled || isLoading}
    className={`flex flex-row items-center px-4 py-2 bg-black text-white rounded-md hover:scale-105 duration-300 ${
      className || ""
    } ${disabled ? "bg-gray-400" : ""}`}
    {...props}
  >
    {isLoading && (
      <svg
        className="animate-spin h-5 w-5 mr-3 text-white"
        fill="none"
        viewBox="0 0 24 24"
      >
        <circle
          className="opacity-25"
          cx="12"
          cy="12"
          r="10"
          stroke="currentColor"
          strokeWidth="4"
        />
        <path
          className="opacity-75"
          fill="currentColor"
          d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        />
      </svg>
    )}
    {text}
  </button>
);

export default Button;
