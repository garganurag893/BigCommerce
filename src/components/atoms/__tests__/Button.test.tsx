import { render, fireEvent } from '@testing-library/react'

import Button from '../Button'

const sampleText = 'This is my first test'
const sampleClassName = 'testClass'

describe('Test Button Component', () => {
  it('Check for button tag in DOM', () => {
    const onClickHandler = jest.fn()
    const { getByRole, getByText } = render(
      <Button text={sampleText} onClick={onClickHandler} className={sampleClassName} />,
    )

    expect(getByRole('button')).toBeInTheDocument()
    expect(getByText(sampleText)).toBeInTheDocument()
    expect(getByRole('button')).toHaveClass(sampleClassName)

    fireEvent.click(getByRole('button'))

    expect(onClickHandler).toHaveBeenCalledTimes(1)
  })

  it('Button Snapshot Test', () => {
    const onClickHandler = jest.fn()
    const { container } = render(<Button text={sampleText} onClick={onClickHandler} />)
    expect(container).toMatchSnapshot()
  })

  it('Button Disabled Snapshot Test', () => {
    const onClickHandler = jest.fn()
    const { container } = render(
      <Button text={sampleText} disabled onClick={onClickHandler}>
        {sampleText}
      </Button>,
    )
    expect(container).toMatchSnapshot()
  })

  it('Button Loading Snapshot Test', () => {
    const onClickHandler = jest.fn()
    const { container } = render(<Button text={sampleText} isLoading onClick={onClickHandler} />)
    expect(container).toMatchSnapshot()
  })
})
