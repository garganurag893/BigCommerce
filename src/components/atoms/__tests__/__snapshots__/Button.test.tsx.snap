// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component Button Disabled Snapshot Test 1`] = `
<div>
  <button
    class="flex flex-row items-center px-4 py-2 bg-black text-white rounded-md hover:scale-105 duration-300  bg-gray-400"
    disabled=""
  >
    This is my first test
  </button>
</div>
`;

exports[`Test Button Component Button Loading Snapshot Test 1`] = `
<div>
  <button
    class="flex flex-row items-center px-4 py-2 bg-black text-white rounded-md hover:scale-105 duration-300  "
    disabled=""
  >
    <svg
      class="animate-spin h-5 w-5 mr-3 text-white"
      fill="none"
      viewBox="0 0 24 24"
    >
      <circle
        class="opacity-25"
        cx="12"
        cy="12"
        r="10"
        stroke="currentColor"
        stroke-width="4"
      />
      <path
        class="opacity-75"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
    This is my first test
  </button>
</div>
`;

exports[`Test Button Component Button Snapshot Test 1`] = `
<div>
  <button
    class="flex flex-row items-center px-4 py-2 bg-black text-white rounded-md hover:scale-105 duration-300  "
  >
    This is my first test
  </button>
</div>
`;
